rust/agama-network/src/adapter.rs

```diff
@@ -0,0 +1,8 @@
+use crate::NetworkState;
+use std::error::Error;
+
+/// A trait for the ability to read/write from/to a network service
+pub trait Adapter {
+    fn read(&self) -> Result<NetworkState, Box<dyn Error>>;
+    fn write(&self, network: &NetworkState) -> Result<(), Box<dyn Error>>;
```

Comment:
to be honest I am not big fan of that box dyn errors...lets hope that with anyerror it can be just anyError